### PR TITLE
Adds labels and annotations from func.yaml to pipeline resources

### LINKED ISSUE #1192: On-cluster build: Apply annotations and labels from `func.yaml` to all created Pipeline resources

### PR DESCRIPTION
**Pull request by gab-satchi**
# Changes

- 🎁 labels from func.yaml are added to Pipeline resources

/kind enhancement

Fixes #1192 

**Release Note**

```release-note
Labels and annotations from func.yaml are added to Pipeline resources
```

### CODE REVIEW

**Review by zroubalik — COMMENTED** [review]
Thanks for contributing this.

We should unify this with labels that are already applied to KSV during the standard deployment, it can also process local env variables:
https://github.com/knative-sandbox/kn-plugin-func/blob/b117af7c9237326eef83cef61e252f3da6f7d6be/knative/deployer.go#L404

**Review by zroubalik — COMMENTED** [review]
Also, https://github.com/knative-sandbox/kn-plugin-func/issues/1192 is about adding annotations as well.

**Comment by gab-satchi** [comment]
Thanks, will update

**Comment by gab-satchi** [comment]
Seems to be another irrelevant error:
```
The Service "kourier-ingress" is invalid: spec.ports[0].nodePort: Invalid value: 30080: provided port is already allocated
Error: Process completed with exit code 1.
```

/retest

File: function.go [hunk 1]
```diff
@@ -344,6 +346,128 @@ func (f Function) ImageWithDigest() string {
...
+// registry:  docker.io/myname
+//            myname
+// A full image name consists of registry, image name and tag.
+// in form [registry]/[image-name]:[tag]
+// example docker.io/alice/my.example.func:latest
+// Default if not provided is --registry (a required global setting)
+// followed by the provided (or derived) image name.
+// TODO: this calculated field should probably be generated on instantiation
+// to avoid confusion.
+func DerivedImage(root, registry string) (image string, err error) {
```
Comment (zroubalik) [inline function.go:417]: This function is probably a result of bad merge, it shouldn't be here. @lkingland refactored this functionality.

**Review by zroubalik — COMMENTED** [review]
Looking good! Just one nit.

File: function.go (hunk 1 above)
Comment (gab-satchi) [reply function.go]: Yes it was, thanks. Removed.

**Review by zroubalik — APPROVED** [review]
/lgtm
/unhold

Great job!

**Pull request merged**